compiler/basicTypes/Weight.hs

```diff
@@ -121,12 +119,12 @@ instance Num (GMult a) where
   (+) = RigAdd
 
 instance Multable a => Outputable (GMult a) where
-  ppr Zero = text "0"
-  ppr One = text "1"
-  ppr Omega = text "ω"
-  ppr (RigAdd m1 m2) = parens (ppr m1 <+> text "+" <+> ppr m2)
-  ppr (RigMul m1 m2) = parens (ppr m1 <+> text "*" <+> ppr m2)
-  ppr (RigThing t) = ppr t
+  ppr Zero_ = text "0"
+  ppr One_ = text "1"
+  ppr Omega_ = text "ω"
+  ppr (RigAdd_ m1 m2) = parens (ppr m1 <+> text "+" <+> ppr m2)
+  ppr (RigMul_ m1 m2) = parens (ppr m1 <+> text "*" <+> ppr m2)
+  ppr (RigThing_ t) = ppr t
```

Comment:
Why are the pattern synonyms wrong here? I'd rather use them: they're prettier.

Reply:
Fixed. Pattern synonyms were giving an incorrect incomplete pattern matching warning, and I wanted to silence it. I forgot that the COMPLETE pragma can be used for this.

Reply:
Ah. That's my bad. I somehow thought I'd done that already.